docs/devel/api_changes.md

```diff
@@ -292,6 +292,33 @@ consumers, please send an announcement to `kubernetes-dev@googlegroups.com` befo
 the change gets in. If you are unsure, ask. Also make sure that the change gets documented in
 `CHANGELOG.md` for the next release.
 
+## Defining new types
+
+Prefer to group together related concepts into a sub-struct, rather than having lots of fields at
```

Comment:
We do this in quite a few places:
* resources
* lifecycle
* capabilities
* security context

It helps to reduce the apparent surface area of the API, especially where the entire group of fields can be set automatically or avoided entirely in most cases. I can say from experience that it's better than a flat structure with dozens of fields with unclear relationships to each other.

Obviously we shouldn't go overboard, such as putting every pair of fields into a subobject. For instance, I wouldn't put Command and Args into a subobject.